controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/zone/Zones.java

```diff
@@ -0,0 +1,46 @@
+package com.yahoo.vespa.hosted.controller.api.integration.zone;
+
+import com.yahoo.config.provision.Environment;
+import com.yahoo.config.provision.RegionName;
+
+/**
+ * A ZoneId list which can be filtered in various ways; elements can be accessed after at least one filter.
+ *
+ * The class is split in an outer and an inner class to force the user to make an initial choice between
+ * the filters available in the outer class; these should be pertinent to what code should use the zone,
+ * while the filters in the inner class are optional and / or for convenience.
+ *
+ *
+ * @author jvenstad
+ */
+public interface Zones {
+
+    /** Negates the next filter. */
+    Zones not();
+
+    /** All zones from the initial pool. */
+    Zones.List all();
+
+    /** Zones where which are managed by the controller. */
+    Zones.List controllerManaged();
+
+
+    /** Wraps access to the zones; this forces the user to consider which zones to access. */
+    interface List extends Zones {
```

Comment:
Not a fan of nested interfaces. I suggest pulling this up to a separate interface named `ZoneList` or `ZoneIdList`, this would be consistent with how we wrap `List<T>` elsewhere, e.g. `ApplicationList` in this module and `com.yahoo.vespa.hosted.provision.NodeList`.

Reply:
I'll try that out and see how it looks.